src/sentry/integrations/slack/notify_action.py

```diff
@@ -0,0 +1,141 @@
+from __future__ import absolute_import
+
+from django import forms
+from django.utils.translation import ugettext_lazy as _
+
+from sentry import http
+from sentry.rules.actions.base import EventAction
+from sentry.utils import metrics, json
+from sentry.models import Integration
+
+from .utils import build_attachment
+
+
+class SlackNotifyServiceForm(forms.Form):
+    team = forms.ChoiceField(choices=(), widget=forms.Select(
+        attrs={'style': 'width:150px'},
+    ))
+    channel = forms.CharField(widget=forms.TextInput(
+        attrs={'placeholder': 'i.e #critical-errors'},
+    ))
+
+    def __init__(self, *args, **kwargs):
+        team_list = [(i.id, i.name) for i in kwargs.pop('integrations')]
+        self.channel_transformer = kwargs.pop('channel_transformer')
+
+        super(SlackNotifyServiceForm, self).__init__(*args, **kwargs)
+
+        self.fields['team'].choices = team_list
+        self.fields['team'].widget.choices = self.fields['team'].choices
+
+    def clean_channel(self):
+        channel = self.cleaned_data.get('channel', '').lstrip('#')
+        team = self.cleaned_data.get('team')
+
+        channel_id = self.channel_transformer(team, channel)
+
+        if channel_id is None and team is not None:
+            params = {
+                'channel': channel,
+                'team': dict(self.fields['team'].choices).get(int(team)),
+            }
+
+            raise forms.ValidationError(
+                _('The #%(channel)s channel does not exist in the %(team)s Slack team.'),
+                code='invalid',
+                params=params,
+            )
+
+        return channel
+
+
+class SlackNotifyServiceAction(EventAction):
+    form_cls = SlackNotifyServiceForm
+    label = 'Send a notification to the Slack {team} team in {channel}'
+
+    def is_enabled(self):
+        return self.get_integrations().exists()
+
+    def after(self, event, state):
+        channel = self.get_option('channel')
+        try:
+            integration = Integration.objects.get(
+                provider='slack',
+                organizations=self.project.organization,
+            )
+        except Integration.DoesNotExist:
```

Comment:
There's nothing I'm seeing that restricts this to just 1 per org? So could this raise for `Integrations.MultipleObjectsReturned`

`get_integrations()` is also seeming to imply that there may be more than 1. The only UNIQUEness on the database side is `(provider, external_id)`

Reply:
You're right, this needs scoped down to the specific integration the rule is triggering for. will fix

Reply:
d05cfffb6f4c2aa2bd919e89ff7e8f9e148a2e01